src/topology/local_homeomorph.lean

```diff
@@ -589,6 +589,69 @@ def to_homeomorph_of_source_eq_univ_target_eq_univ (h : e.source = (univ : set 
 @[simp] lemma to_homeomorph_inv_fun (h : e.source = (univ : set α)) (h' : e.target = univ) :
   (e.to_homeomorph_of_source_eq_univ_target_eq_univ h h').inv_fun = e.inv_fun := rfl
 
+section curry
+
+open_locale classical
+
+/-- Given a local homeomorphism between `α × β` and ‵γ` and a point `x : α` such that
+`(x, y) ∈ e.source` for some `y`, `e.curry x hx` is a local homeomorphism between
+`β` and `{z | (e⁻¹ z).1 = x}`. Varying `x`, we get a family of local homeomorphisms
+that can be interpreted as a version of `function.curry` for `local_homeomorph`s.
+Note that in contrast with `function.curry` this operation looses information,
+so there is no `local_homeomorph.uncurry`.  -/
```

Comment:
Could you add in the docstring an explanation why you need to restrict the target type (i.e., replace gamma by a subtype): otherwise, the target wouldn't be open in the target type.

Reply:
BTW, will it look better if I swap `e` with `e.symm` and call it `fiber_homeomorph`?

Reply:
`fiber_homeomorph` would be great. As for swapping, I don't have an opinion, both conventions make sense. So do as you prefer.